### DISCUSSION TITLE
How to copy a Formula cell and increment formula references automatically?

### DISCUSSION THREAD

**Discussion by alvarovillafane** [handsontable/hyperformula#1171] [API Questions]
I would like to know if there is a way to copy formulas with Hyperformula similar to how you would in Excel by dragging a cell down and having the formulas update automatically.

For example, I have a complex formula that includes references to other sheets and cells, as well as fixed references (with the $ sign). The formula looks something like this:

`=Sheet2!A2 + A3 + TIMEVALUE("15:00") + Sheet2!C:C + Sheet2!$A1 + B3 + C3 + Sheet2!A$2`

When I drag this formula down to the next cell, I want it to update automatically to reflect the new cell references. In this case, the formula would become:

`=Sheet2!A3 + A4 + TIMEVALUE("15:00") + Sheet2!C:C + Sheet2!$A2 + B4 + C4 + Sheet2!A$2`

Can anyone suggest if this is supported or have any idea on how to do this in an efficient way? 

Thanks!

**Comment by AMBudnik** [comment]
Hi @vmalvaro 

It seems that we have a similar topic here https://github.com/handsontable/hyperformula/issues/887

Would that be helpful and meet the project requirements?

**Comment by sequba** [comment]
@vmalvaro take a look at [getFillRangeData](https://hyperformula.handsontable.com/api/classes/hyperformula.html#getfillrangedata) method and discussion https://github.com/handsontable/hyperformula/discussions/1149

**Comment by alvarovillafane** [comment]
yeah, that's awesome, thanks @AMBudnik @sequba!!
